### DISCUSSION TITLE
how to create a .mdpa file?

### DISCUSSION THREAD

**Discussion by Ada2045lalala** [KratosMultiphysics/Kratos#12113] [Q&A] [answered]
I am trying to use the ShapeOptimizationApplication just like [this example.](https://github.com/KratosMultiphysics/Examples/tree/master/shape_optimization/use_cases/11_Shape_Update_Optimization_Stanford_Bunny)
And there is a bunny.mdpa file in the folder. I am wondering **how to create a .mdpa file** which contains the information of a bounding geometry. (I have tried to get the answer in wiki,but failed.) And how to convert a bounding geometry in **another format to mdpa format**?

**Comment by loumalouomega** [comment]
Right now the easiest way to generate a mdpa files is to use [GiD](https://www.gidsimulation.com/) or [Salome plugin](https://github.com/KratosMultiphysics/KratosSalomePlugin). Another alternative, but at the moment lacks supports is using [meshio](https://github.com/nschloe/meshio). Other alternatives require more work and are not trivial.

Other input are not currently supported at the moment, but there is a plan to expand it at some point in the mid term.

**Reply by Ada2045lalala** [reply]
Thank you for your response!
I visited GiD's website, and have a few questions.

1. how to use GiD to generate such a complicated geometry like this bunny?
![bunny_pic(1)](https://github.com/KratosMultiphysics/Kratos/assets/155991150/0341901a-b2c9-40e0-8234-63d6a077e0c5)

2.could [this course](https://www.gidsimulation.com/gid-for-science/support/courses/) enables me to generate a geometry like the bunny above?
![GiD course](https://github.com/KratosMultiphysics/Kratos/assets/155991150/2691733c-d4d1-4659-9bb2-4c922d2ca1c4)

![course_content](https://github.com/KratosMultiphysics/Kratos/assets/155991150/f8ab064a-f0c2-4531-93f5-c2eb326ce226)

**Reply by rubenzorrilla** [reply]
@jginternational

**Reply by jginternational** [reply]
hey @Ada2045lalala 
You can import the bunny mesh in the format you want. GiD imports  a lot of mesh and geometry formats.
https://gidsimulation.atlassian.net/wiki/spaces/GUM/pages/2385552360/Importing+on+GiD
Check your GiD in the menú Files > Import.

Once imported the mesh, you can create the surface geometry using the menu
Geometry > Create > Geometry from mesh > One element - one surface . Then select the triangles and escape

After that you can create a volume using the closed surfaces, and assign it to your kratos parts, to generate a mdpa.

Remember to load Kratos using the Menu Data > Problemtype > Kratos

Hope it helps

**Reply by loumalouomega** [reply]
Mesh

**Reply by Ada2045lalala** [reply]
thanks for your help.@jginternational 
I'll try your suggestion as soon as I complete mesh.

**Reply by Ada2045lalala** [reply]
hi @jginternational 
**I meet questions again, and I need your help. Thanks a lot!** 

I try to generate a mdpa file as follows. However, fail.

1. draw a cube surface like this:![image](https://github.com/KratosMultiphysics/Kratos/assets/155991150/350eba57-a7eb-467f-b728-2cab9ddabdc0)
2. mesh:![image](https://github.com/KratosMultiphysics/Kratos/assets/155991150/4db50183-0c80-40d3-aed8-965f0e14a4dc)

3. switch to kratos mdpa generator:
![image](https://github.com/KratosMultiphysics/Kratos/assets/155991150/bb96dc7b-b874-4787-9f6c-0d2c22a27df2)
![image](https://github.com/KratosMultiphysics/Kratos/assets/155991150/f80eb5c0-d904-421d-b8c2-70bd03f164eb)

4. files>>export>>meshio...
![image](https://github.com/KratosMultiphysics/Kratos/assets/155991150/db55d7f9-aa69-424c-9bbf-850b6d98d846)

5. here is the error info:
![image](https://github.com/KratosMultiphysics/Kratos/assets/155991150/f37abe37-60cc-4741-b0bd-234f1b27a348)

**Comment by jginternational** [comment, marked as answer]
Ok, steps 1 and 3 are fine.

Once you load the Model part application, assign the entities you need to Submodelparts, and select the writing mode (maybe you just need the nodes, or the elements...)

![fill1](https://github.com/KratosMultiphysics/Kratos/assets/5918085/23a7e8f0-055a-4cf9-bef1-eea43dbdc9be)

Once that is done, generate the mesh like in step 2, and save the model.

![sni](https://github.com/KratosMultiphysics/Kratos/assets/5918085/7f226328-3866-471c-9951-ffddcf48377d)



To export the mdpa just click on calculate (menu calculate or the rocket in the toolbar)
![res](https://github.com/KratosMultiphysics/Kratos/assets/5918085/ba447b51-a38f-4588-851a-dc21151b40bb)

Change the parameters in the app to export as element& conditions or geometries
